src/Features/VisualBasic/Portable/Completion/CompletionProviders/SymbolCompletionProvider.vb

```diff
@@ -20,6 +20,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.Providers
             Return Recommender.GetRecommendedSymbolsAtPositionAsync(context.SemanticModel, position, context.Workspace, options, cancellationToken)
         End Function
 
+        Protected Overrides Async Function GetPreselectedSymbolsWorker(context As AbstractSyntaxContext, position As Integer, options As OptionSet, cancellationToken As CancellationToken) As Task(Of IEnumerable(Of ISymbol))
+            Dim typeInferrer = context.GetLanguageService(Of ITypeInferenceService)()
+            Dim inferredTypes = typeInferrer.InferTypes(context.SemanticModel, position, cancellationToken).ToSet()
+            If inferredTypes Is Nothing OrElse Not inferredTypes.Any() Then
+                Return SpecializedCollections.EmptyEnumerable(Of ISymbol)()
+            End If
+
+            Dim symbols = Await Recommender.GetRecommendedSymbolsAtPositionAsync(context.SemanticModel, position, context.Workspace, options, cancellationToken).ConfigureAwait(False)
+            Return symbols.Where(Function(s) inferredTypes.Contains(s.GetSymbolType()))
```

Comment:
TitTity                                                                                                                                                                                                                From: CyrusNajmabadiSent: Tuesday 23 February 2016 15:28To: dotnet/roslynReply To: dotnet/roslynSubject: Re: [roslyn] Implement "Target type preselection" (#6878)In src/Features/VisualBasic/Portable/Completion/CompletionProviders/SymbolCompletionProvider.vb:
> @@ -20,6 +20,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.Providers
>              Return Recommender.GetRecommendedSymbolsAtPositionAsync(context.SemanticModel, position, context.Workspace, options, cancellationToken)
>          End Function
>  
> +        Protected Overrides Async Function GetPreselectedSymbolsWorker(context As AbstractSyntaxContext, position As Integer, options As OptionSet, cancellationToken As CancellationToken) As Task(Of IEnumerable(Of ISymbol))
> +            Dim typeInferrer = context.GetLanguageService(Of ITypeInferenceService)()
> +            Dim inferredTypes = typeInferrer.InferTypes(context.SemanticModel, position, cancellationToken).ToSet()
> +            If inferredTypes Is Nothing OrElse Not inferredTypes.Any() Then
> +                Return SpecializedCollections.EmptyEnumerable(Of ISymbol)()
> +            End If
> +
> +            Dim symbols = Await Recommender.GetRecommendedSymbolsAtPositionAsync(context.SemanticModel, position, context.Workspace, options, cancellationToken).ConfigureAwait(False)
> +            Return symbols.Where(Function(s) inferredTypes.Contains(s.GetSymbolType()))

You didn't do the intrinsic check here.

—Reply to this email directly or view it on GitHub.